src/js/homepage.js

```diff
@@ -2,6 +2,102 @@ import React from "react";
 import ReactDOM from "react-dom";
 import _ from "lodash";
 
+class TwitterTabs extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      timelineTab: 'timelineTab tab',
+      userTweetsTab: 'userTweetsTab tab',
+      postTweetTab: 'postTweetTab tab',
+    }
+  }
+
+  openTab(tabId) {
+    let tabContent = document.getElementsByClassName('tabContent');
+    _.forEach(tabContent, content => {
+      content.style.display = 'none';
+    });
+
+    this.setState({
+      timelineTab: 'timelineTab tab',
+      userTweetsTab: 'userTweetsTab tab',
+      postTweetTab: 'postTweetTab tab',
+    });
+
+    switch(tabId) {
+      case('timelineColumn'):
+        this.setState({
+          timelineTab: 'timelineTab tab active'
+        });
+        break;
+      case('userTweetsColumn'):
+        this.setState({
+          userTweetsTab: 'userTweetsTab tab active'
+        });
+        break;
+      case('postTweetColumn'):
+        this.setState({
+          postTweetTab: 'postTweetTab tab active'
+        });
+    }
+
+    document.getElementById(tabId).style.display = 'block';
+  }
+
+  render() {
+    return React.createElement('div', {className: 'tabDiv'},
+      React.createElement('button', {className: this.state.timelineTab, id: 'defaultTab', onClick: () => this.openTab('timelineColumn')}, 'Home Timeline'),
```

Comment:
nice job getting it working, the current implementation is a bit brittle since the entire class needs to be preserved across the entire state